src/storage/src/table/storage_table.rs

```diff
@@ -603,7 +569,7 @@ impl<S: StateStore, E: Encoding, const T: AccessType> StorageTableBase<S, E, T>
                     } else {
                         // Should use excluded next key for end bound.
                         // Otherwise keys starting with the bound is not included.
-                        Excluded(next_key(&serialized_key))
+                        end_bound_of_prefix(&serialized_key)
```

Comment:
Here we also fix that, if `serialized_key` is `\xff`, we should return `Unbounded` here instead of `Excluded(b"")`. cc @xxchan